### PR TITLE
Operator: Add missing observability for Azure API calls

### PR DESCRIPTION
**Pull request by hemanthmalla**
Currently we're not capturing response status and duration for some of the calls made to Azure. This commit calls `ObserveAPICall()` with status and duration for the operations missing them.

### CODE REVIEW

**Comment by hemanthmalla** [comment]
@tommyp1ckles / @gandro Gentle nudge, can I get a quick review on this ? 
CI is green and the PR only adds some missing observability. Thank you.

File: pkg/azure/api/api.go [hunk 1]
```diff
@@ -438,22 +440,27 @@ func (c *Client) AssignPrivateIpAddressesVMSS(ctx context.Context, instanceID, v
 	}
 
 	c.limiter.Limit(ctx, "VirtualMachineScaleSetVMs.Update")
+	sinceStart = spanstat.Start()
 	future, err := c.vmss.Update(ctx, c.resourceGroup, vmssName, instanceID, result)
 	if err != nil {
+		c.metricsAPI.ObserveAPICall("VirtualMachineScaleSetVMs.Update", deriveStatus(err), sinceStart.Seconds())
```
Comment (gandro) [inline pkg/azure/api/api.go]: nit: Have you considered using defer for these?

**Review by gandro — APPROVED** [review]

**Review by tommyp1ckles — APPROVED** [review]
looks good - same question about the defer

File: pkg/azure/api/api.go (hunk 1 above)
Comment (hemanthmalla) [reply pkg/azure/api/api.go]: Good idea. Updated this.

**Pull request merged**